Composer/packages/server/src/models/bot/botProject.ts

```diff
@@ -433,12 +435,21 @@ export class BotProject {
     if (index === -1) {
       throw new Error(`no such file at ${relativePath}`);
     }
-
     const absolutePath = `${this.dir}/${relativePath}`;
-    await this.fileStorage.writeFile(absolutePath, content);
+
+    // only write if the file has actually changed
+    if (this.files[index].content !== content) {
+      await this.fileStorage.writeFile(absolutePath, content);
+    }
+
+    // TODO: we should get the lastmodified from the writeFile operation
```

Comment:
s/lastmodified/lastModified

Reply:
speaking to the TODO comment, perhaps we can _memoize_ `stat` to avoid redundant storage operations